pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -268,3 +305,18 @@ func fitsRequest(podRequest *preFilterState, nodeInfo *framework.NodeInfo, ignor
 
 	return insufficientResources
 }
+
+func isIgnoredResource(name string, ignoredResources, ignoredGroups sets.String) bool {
+	if ignoredResources.Has(name) {
+		return true
+	}
+
+	for group := range ignoredGroups {
+		// We know that name is the extended resource, so it is divided by "/".
+		// group matches the substring before "/".
+		if strings.HasPrefix(name, group+"/") {
```

Comment:
This will allocate a new string for every ignoredGroup on every invocation of isIgnoredResource.

to avoid allocations in this inner loop, it would be better to do the following:
* ignore names that don't contain a `/`
* get a subslice of name prior to `/`
* check ignoredGroups.Has(nameBeforeSlash)

Reply:
could you review it again?

Reply:
What if there are multiple slashes? Is that a valid label?

@angao can you confirm if doing .Split doesn't allocate new strings too?

Reply:
As we know, the resource is the [extended resource,](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#extended-resources) so its format must be `domain.io/aaa`, so `domain.io/aaa/bbb` is invalid and `domain.io//` is illegal.

From the above,  if the resource names are `example.com/aaa` and `example.com/bbb`, and we want to ignore all resources, we can set the `ignoredGroups` is `example.com`, and `example.com/` is invalid, `example` is not satisfying the needs.

@alculquicondor  WDYT？